src/genie/libs/parser/ios/tests/test_show_ospf.py

```diff
@@ -0,0 +1,224 @@
+class test_show_ip_ospf_neighbor_detail(test_show_ip_ospf_neighbor_detail_iosxe):
+
+    def test_show_ip_ospf_neighbor_detail_full1(self):
+        super().test_show_ip_ospf_neighbor_detail_full1()
+        self.maxDiff = None
+
+        def mapper(key):
+            return self.outputs[key]
+
+
+        self.device.execute = Mock()
+        self.device.execute.side_effect = mapper
+
+        obj = ShowIpOspfNeighborDetail(device=self.device)
+        parsed_output = obj.parse()
+        self.assertEqual(parsed_output, self.golden_parsed_output1)
+
+    def test_show_ip_ospf_neighbor_detail_full2(self):
+        super().test_show_ip_ospf_neighbor_detail_full2()
+
+        self.maxDiff = None
+
+        def mapper(key):
+            return self.outputs[key]
+
+        self.device.execute = Mock()
+        self.device.execute.side_effect = mapper
+
+        obj = ShowIpOspfNeighborDetail(device=self.device)
+        parsed_output = obj.parse()
+        self.assertEqual(parsed_output, self.golden_parsed_output2)
+
+    def test_show_ip_ospf_neighbor_detail_full3(self):
+        super().test_show_ip_ospf_neighbor_detail_full3()
+
+        self.maxDiff = None
+
+        def mapper(key):
+            return self.outputs[key]
+
+
+        self.device.execute = Mock()
+        self.device.execute.side_effect = mapper
+        obj = ShowIpOspfNeighborDetail(device=self.device)
+        parsed_output = obj.parse()
+        self.assertEqual(parsed_output, self.golden_parsed_output3)
+
+    def test_show_ip_ospf_neighbor_detail_empty(self):
+        self.maxDiff = None
+        self.device = Mock(**self.empty_output)
+        obj = ShowIpOspfNeighborDetail(device=self.device)
+        with self.assertRaises(SchemaEmptyParserError):
+            parsed_output = obj.parse()
+
+
+# =======================================
+# Unit test for 'show ip ospf sham-links'
+# =======================================
+class test_show_ip_ospf_sham_links(test_show_ip_ospf_sham_links_iosxe):
+
+    def test_show_ip_ospf_sham_links_full1(self):
+        super().test_show_ip_ospf_sham_links_full1()
+        self.maxDiff = None
+
+        def mapper(key):
+            return self.outputs[key]
+
+        self.device.execute = Mock()
+        self.device.execute.side_effect = mapper
+
+        obj = ShowIpOspfShamLinks(device=self.device)
+        parsed_output = obj.parse()
+        self.assertEqual(parsed_output, self.golden_parsed_output1)
+
+    def test_show_ip_ospf_sham_links_empty(self):
+        self.maxDiff = None
+        self.device = Mock(**self.empty_output)
+        obj = ShowIpOspfShamLinks(device=self.device)
+        with self.assertRaises(SchemaEmptyParserError):
+            parsed_output = obj.parse()
+
+
+# ==========================================
+# Unit test for 'show ip ospf virtual-links'
+# ==========================================
+class test_show_ip_ospf_virtual_links(test_show_ip_ospf_virtual_links_iosxe):
+
+    golden_parsed_output1 = {
```

Comment:
Hi @itdependsnetworks ,

Seems both old and new ways of creating unit tests are included in this PR.

Reply:
honestly, it is difficult for me to track what the mapper is actually doing. Are they needed? is the any ability for someone to convert them?